S2VX.Game/SongSelection/UserInterface/BorderOuterBox.cs

```diff
@@ -0,0 +1,30 @@
+﻿using osu.Framework.Allocation;
+using osu.Framework.Input.Events;
+using osu.Framework.Screens;
+using osuTK.Graphics;
+using S2VX.Game.Play;
+
+namespace S2VX.Game.SongSelection.UserInterface {
+    public class BorderOuterBox : RelativeBox {
+        [Resolved]
+        private ScreenStack Screens { get; set; }
+
+        public SongSelectionScreen SongSelectionScreen { get; set; }
+
+        // Is called by InnerBox when mouse hovers over InnerBox (and thus should not be considered over OuterBox)
+        public void LoseHover() => Colour = Color4.White;
+
+        protected override bool OnHover(HoverEvent e) {
+            Colour = Color4.LightGray;
+            return false;
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e) => LoseHover();
```

Comment:
Instead of having the LoseHover() function, wouldn't it work if you just did everything in this function? The LoseHover call in BorderInnerBox is also unnecessary I believe, since OnHoverLost would be called in here.

Reply:
You're right, so long as I capture hover inside BorderInnerBox (return true), it'll act as if hover is lost in BorderOuterBox.